pan-domain-auth-verification/src/main/scala/com/gu/pandomainauth/service/CryptoConf.scala

```diff
@@ -23,9 +24,22 @@ object CryptoConf {
 
     lazy val acceptedPublicKeys: LazyList[PublicKey] = LazyList(activePublicKey) ++ alsoAccepted
 
+    private lazy val activeKeyId = KeyHashId.calculateFor(activePublicKey)
+
+    private lazy val acceptedKeysWithIds: LazyList[(PublicKey, KeyHashId)] =
+      acceptedPublicKeys.map(key => key -> KeyHashId.calculateFor(key))
+
     private[CryptoConf] def acceptsActiveKeyFrom(other: Verification): Boolean = acceptedPublicKeys.contains(other.activePublicKey)
 
-    def decode[A](f: PublicKey => Option[A]): Option[A] = acceptedPublicKeys.flatMap(f(_)).headOption
+    def decode[A](f: PublicKey => Option[A]): Option[A] = {
+      (for {
+        (key, keyId) <- acceptedKeysWithIds
+        result <- f(key)
+      } yield {
+        NonActiveKeyMonitoring.instance.monitor(keyId, activeKeyId)
```

Comment:
This is the essential change in this PR - we call out to a logging class, passing enough information for it to record key usage, and record if the key is active - or not.